pkg/foundation/cerrors/fatal.go

```diff
@@ -0,0 +1,41 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cerrors
+
+// FatalError is an error type that will diferentiate these from other errors that could be retried.
+type FatalError struct {
+	Err error
+}
+
+// NewFatalError creates a new FatalError.
+func NewFatalError(err error) *FatalError {
+	return &FatalError{Err: err}
+}
+
+// Unwrap returns the wrapped error.
+func (f *FatalError) Unwrap() error {
+	return f.Err
+}
+
+// Error returns the error message.
+func (f *FatalError) Error() string {
+	return f.Err.Error()
+}
+
+// IsFatalError checks if the error is a FatalError.
+func IsFatalError(err error) bool {
+	_, ok := err.(*FatalError)
```

Comment:
after some research, this check wouldn't work if the fatalError is wrapped, something like 
> err := fmt.Errorf("something went wrong: %w", NewFatalError(errors.New("fatal error")))

will return false for this method.
looks like `errors.As` should be used instead, which is just `As` in this package

Reply:
Same as with https://github.com/ConduitIO/conduit/pull/1811#discussion_r1735998845, this would require importing the `errors` package which is not allowed:

```
pkg/foundation/cerrors/fatal.go:18:2: import 'errors' is not allowed from list 'main': To accommodate for custom error creation and handling use Conduit's 'cerrors' package instead. (depguard)
	"errors"
```

I'll check what else I can do.

Reply:
Oh! I misunderstood what you said @maha-hajja 🤦 . I see that `As` is actually defined in the `cerrors` pkg, and that's exactly what you meant! 😓 . I'll make that change. Thanks for the pointer.

Reply:
@maha-hajja pushed a fix https://github.com/ConduitIO/conduit/pull/1811/commits/e812e1c64655e000cc6df350c3f05c7fb59d88aa